### PR TITLE
Flip filesinrepos to true if false

### PR DESCRIPTION
**Pull request by pritishpai**

### CODE REVIEW

File: src/databricks/labs/blueprint/installation.py [hunk 1]
```diff
@@ -169,6 +169,7 @@ def load(self, type_ref: type[T], *, filename: str | None = None) -> T:
         the expected version using a method named `v{actual_version}_migrate` on the `type_ref` class. If the migration
         is successful, the method will return the migrated object. If the migration is not successful, the method will
         raise an `IllegalState` exception."""
+        self._enable_files_in_repos()
```
Comment (nfx) [inline src/databricks/labs/blueprint/installation.py:172]: Do it in _load_content and only when the exception is thrown. That way you don't have to change so many unit tests 😉

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/blueprint/installation.py (hunk 1 above)
Comment (pritishpai) [reply src/databricks/labs/blueprint/installation.py]: Function call now during upload only when "FEATURE_DISABLED" error_code is thrown for NotFound. And retried after flipping the value for enableWorkspaceFilesystem.

File: tests/unit/test_installation.py [hunk 2]
```diff
@@ -327,3 +327,33 @@ def test_migrations_broken():
 
     with pytest.raises(IllegalState):
         installation.load(BrokenConfig)
+
+
+def mock_get_status(*args):
+    random_return_value = random.choice(["true", "false"])
```
Comment (nfx) [inline tests/unit/test_installation.py:333]: don't do random in unit tests. unit tests have to be deterministic

File: tests/unit/test_installation.py [hunk 3]
```diff
@@ -327,3 +327,33 @@ def test_migrations_broken():
...
+    if args[0] == "enableWorkspaceFilesystem":
+        return {"enableWorkspaceFilesystem": random_return_value}
+
+
+def test_enable_files_in_repos(mocker):
+    ws = create_autospec(WorkspaceClient)
+    ws.current_user.me().user_name = "foo"
+    installation = Installation(ws, "ucx")
+
+    ws.workspace_conf.get_status = mocker.patch(
```
Comment (nfx) [inline tests/unit/test_installation.py:343]: don't monkey-patch code you can mock - just use simple mocking `ws.workspace_conf.get_status.return_value = ...`

**Review by nfx — CHANGES_REQUESTED** [review]
don't do random in unit testing

File: tests/unit/test_installation.py [hunk 4]
```diff
@@ -327,3 +326,30 @@ def test_migrations_broken():
...
+
+
+def test_enable_files_in_repos():
+    ws = create_autospec(WorkspaceClient)
+    ws.current_user.me().user_name = "foo"
+    installation = Installation(ws, "ucx")
+
+    # enableWorkspaceFilesystem is true
+    ws.workspace_conf.get_status.return_value = {"enableWorkspaceFilesystem": "true"}
+    installation._enable_files_in_repos()
```
Comment (nfx) [inline tests/unit/test_installation.py:338]: Don't test private methods directly - mock or assert api calls they make

**Review by nfx — CHANGES_REQUESTED** [review]

File: tests/unit/test_installation.py (hunk 4 above)
Comment (pritishpai) [reply tests/unit/test_installation.py:338]: I have asserted the api call from inside _enable_files_in_repos(), but there is still a call to the private method.

**Review by nfx — APPROVED** [review]

**Pull request merged**